packages/viewer/src/modelViewer.js

```diff
@@ -1,98 +1,64 @@
 /* eslint-disable import/no-unresolved */
 import * as THREE from './three/build/three.module.js';
-import { GLTFLoader } from './three/examples/jsm/loaders/GLTFLoader.js';
 import { OrbitControls } from './three/examples/jsm/controls/OrbitControls.js';
-import { Constants } from './motion-controllers.module.js';
+import { VRButton } from './three/examples/jsm/webxr/VRButton.js';
 /* eslint-enable */
 
-import ErrorLogging from './errorLogging.js';
+import ManualControls from './manualControls.js';
+import ControllerModel from './controllerModel.js';
+import ProfileSelector from './profileSelector.js';
+import AssetError from './assetError.js';
+import MockGamepad from './mocks/mockGamepad.js';
+import MockXRInputSource from './mocks/mockXRInputSource.js';
 
 const three = {};
 let canvasParentElement;
-let activeModel;
+
+let profileSelector;
+let mockControllerModel;
+let isImmersive = false;
 
 /**
- * @description Attaches a small blue sphere to the point reported as touched on all touchpads
- * @param {Object} model - The model to add dots to
- * @param {Object} motionController - A MotionController to be displayed and animated
- * @param {Object} rootNode - The root node in the asset to be animated
+ * Adds the event handlers for VR motion controllers to load the assets on connection
+ * and remove them on disconnection
+ * @param {number} index
  */
-function addTouchDots({ motionController, rootNode }) {
-  Object.keys(motionController.components).forEach((componentId) => {
-    const component = motionController.components[componentId];
-    // Find the touchpads
-    if (component.type === Constants.ComponentType.TOUCHPAD) {
-      // Find the node to attach the touch dot.
-      const touchPointRoot = rootNode.getObjectByName(component.touchPointNodeName, true);
-      if (!touchPointRoot) {
-        ErrorLogging.log(`Could not find touch dot, ${component.touchPointNodeName}, in touchpad component ${componentId}`);
-      } else {
-        const sphereGeometry = new THREE.SphereGeometry(0.001);
-        const material = new THREE.MeshBasicMaterial({ color: 0x0000FF });
-        const sphere = new THREE.Mesh(sphereGeometry, material);
-        touchPointRoot.add(sphere);
-      }
-    }
+function initializeVRController(index) {
+  const vrController = three.renderer.xr.getController(index);
+
+  vrController.addEventListener('connected', async (event) => {
+    const controllerModel = new ControllerModel();
+    vrController.add(controllerModel);
+
+    const motionController = await profileSelector.createMotionController(event.data);
+    await controllerModel.initialize(motionController);
   });
+
+  vrController.addEventListener('disconnected', () => {
+    vrController.remove(vrController.children[0]);
```

Comment:
Answering my own question from above: Yeah, this is probably better than a `clear()` method in this context.